File: Java/ejercicio1_2.java
import java.util.*;

/* Implement an algorithm to determine if a string has all unique characters. What if you
cannot use additional data structures?  */

class Word{
	int MAX_CHAR = 256; 

	boolean uniqueCharacters(String str)
	{
		if (str.length() > MAX_CHAR)
			return false;

		boolean[] chars = new boolean[MAX_CHAR];
		Arrays.fill(chars, false);

		for (int i = 0; i < str.length(); i++) {
			int index = (int)str.charAt(i);

			if (chars[index] == true)
				return false;

			chars[index] = true;
		}

		return true;
	}

	// Driver code
	public static void main(String args[])
	{
		Word obj = new Word();
        Scanner lectura = new Scanner (System.in);
        System.out.print("Ingresa una palabra: ");
        String input = lectura.next();

		if (obj.uniqueCharacters(input))
			System.out.println("Caracteres unicos");
		else
			System.out.println("Caracteres duplicados");
	}
}

File: Phyton/Ejercicio1_2.py
# Implement an algorithm to determine if a string has all unique characters. What if you
# cannot use additional data structures?  

def esUnico(str):
    MAX_CHAR = 256
    n = len(str)

    if(n > MAX_CHAR):
        return False
    
    Chars = [False] * MAX_CHAR

    for i in range(n):
        index = ord(str[i])

        if(Chars[index] == True):
            return False
        Chars[index] = True
    return True

print("INGRESA UNA CADENA ")
cadena = input()   
if(esUnico(cadena)):
    print("Existen caracteres repetidos")   
else:
    print("No existen caractere repetidos")           
